### PR TITLE
Chore: Enhancement

### PR DESCRIPTION
**Pull request by OleksiiBulba**
- Added an exception message to ServiceNotRegisteredException exception;
- Added a property `serviceId` to ServiceNotRegisteredException;
- Added .gitattributes files to avoid unused files in vendor folder;

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+jobs:
+  tests:
+    name: "Tests ${{ matrix.php-version }} deps ${{ matrix.dependency-versions }}"
+    runs-on: ubuntu-22.04
+
+    strategy:
+      fail-fast: false
+      matrix:
+        # normal, highest, non-dev installs
+        php-version: [ '8.1', '8.2' ]
```
Comment (Asisyas) [inline .github/workflows/ci.yaml:16]: php version should be equal 8.2 or higher.

File: composer.json [hunk 2]
```diff
@@ -1,23 +1,43 @@
...
         {
             "name": "Stanislau.Komar",
             "email": "stanislau_komar@epam.com"
         }
     ],
     "require": {
-        "php": ">=8.0",
-        "psr/container": "^2.0",
-        "micro/autowire": "^1"
+        "php": "^8.1 || ^8.2",
```
Comment (Asisyas) [inline composer.json:13]: php version should be equal 8.2 or higher.

**Review by Asisyas — DISMISSED** [review]

File: composer.json (hunk 2 above)
Comment (OleksiiBulba) [reply composer.json]: @Asisyas, done

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (OleksiiBulba) [reply .github/workflows/ci.yaml]: @Asisyas, done

**Pull request merged**